code/data_structures/fenwick_tree/README.md

```diff
@@ -1,3 +1,47 @@
-# Cosmos
+## Binary Indexed Tree
+
+The idea is based on the fact that all positive integers can be represented as sum of powers of 2. For example 19 can be represented as 16 + 2 + 1. Every node of BI Tree stores sum of n elements where n is a power of 2. For example, in the above first diagram for query(), sum of first 12 elements can be obtained by sum of last 4 elements (from 9 to 12) plus sum of 8 elements (from 1 to 8). The number of set bits in binary representation of a number n is O(Logn). Therefore, we traverse at-most O(Logn) nodes in both query() and update() operations. Time complexity of construction is O(nLogn) as it calls update() for all n elements. 
+
+[Credits](http://www.geeksforgeeks.org/binary-indexed-tree-or-fenwick-tree-2/)
```

Comment:
Links, only of standard sources such as Wikipedia, are allowed in the end as a further reading section. Kindly remove such links 👍